File: vb/VS2010/netframework/Builder.vb
Namespace MainDemo
	Friend NotInheritable Class Builder

		Private Sub New()
		End Sub

		Friend Shared Sub Build(ByVal Proj As String)
#If(FRAMEWORK40) Then
			Dim prj = New Microsoft.Build.Evaluation.Project(Proj)
			Try
#Else
			Dim prj As New Microsoft.Build.BuildEngine.Project()
			prj.Load(Proj)
#End If

				prj.SetProperty("DefaultTargets", "Build")
				prj.SetProperty("Configuration", "Debug")
				If Not prj.Build() Then
					Throw New Exception("Error building project: " & Proj & vbLf)
				End If
#If(FRAMEWORK40) Then
			Finally
				Microsoft.Build.Evaluation.ProjectCollection.GlobalProjectCollection.UnloadProject(prj)
			End Try
#End If
		End Sub
	End Class
End Namespace

File: csharp/VS2019/netframework/Builder.cs
using System;

namespace MainDemo
{
    static class Builder
    {
        internal static void Build(string Proj)
        {
#if(FRAMEWORK40)
            var prj = new Microsoft.Build.Evaluation.Project(Proj);
            try
            {
#else
            Microsoft.Build.BuildEngine.Project prj = new Microsoft.Build.BuildEngine.Project();
            prj.Load(Proj);
#endif

                prj.SetProperty("DefaultTargets", "Build");
                prj.SetProperty("Configuration", "Debug");
                if (!prj.Build()) throw new Exception("Error building project: " + Proj + "\n");
#if(FRAMEWORK40)
            }
            finally
            {
                Microsoft.Build.Evaluation.ProjectCollection.GlobalProjectCollection.UnloadProject(prj);
            }
#endif
        }
    }
}
